Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm

```diff
@@ -152,10 +152,25 @@
 
 void RemoteLayerTreeDrawingAreaProxy::commitLayerTree(IPC::Connection& connection, const Vector<std::pair<RemoteLayerTreeTransaction, RemoteScrollingCoordinatorTransaction>>& transactions)
 {
-    [CATransaction begin];
-    for (auto& transaction : transactions)
+    Vector<MachSendRight> sendRights;
+    for (auto& transaction : transactions) {
+        // commitLayerTreeTransaction consumes the incoming buffers, so we need to grab them first.
+        for (auto& [layerID, properties] : transaction.first.changedLayerProperties()) {
+            const RemoteLayerBackingStoreProperties *backingStoreProperties = properties->backingStoreProperties.get();
+            if (!backingStoreProperties)
+                continue;
+            if (const auto& backendHandle = backingStoreProperties->bufferHandle()) {
+                if (const auto* sendRight = std::get_if<MachSendRight>(&backendHandle.value()))
+                    sendRights.append(*sendRight);
+            }
+        }
+
         commitLayerTreeTransaction(connection, transaction.first, transaction.second);
-    [CATransaction commit];
+    }
+
+    // Keep IOSurface send rights alive until the commit makes it to the render server, otherwise we will
+    // prematurely drop the only reference to them, and `inUse` will be wrong for a brief window.
+    [CATransaction addCommitHandler:[sendRights = WTFMove(sendRights)]() { } forPhase:kCATransactionPhasePostSynchronize];
```

Comment:
btw, in [a recent PR](https://github.com/WebKit/WebKit/pull/15552) I had to `mutable` a handler because it copied the capture args rather than moving (when converting from a C++ lambda to an ObjC block). That may or may not be needed here?...

Reply:
Hm...I don't _think_ it's necessary here; in your other PR, the `makeBlockPtr` call (which wraps `Block_copy` in non-ARC) was turning the move into a copy, but in this case it's directly converting the lambda to a block.

(Might be good to double check, however).